### PR TITLE
fix `There are some JS errors on page https://lenta.ru/rubrics/russia/` (close #1191)

### LINKED ISSUE #1191: Wrong port is applied to a cross-domain iframe location after redirect
The page without hammerhead:
image

The page in hammerhead-playground:
image

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
ping

File: src/request-pipeline/header-transforms.js [hunk 1]
```diff
@@ -115,9 +115,15 @@ var responseTransforms = {
         if (!host)
             src = resolveUrl(ctx.dest.url, src);
 
-        var isCrossDomain = ctx.isIframe && !urlUtils.sameOriginCheck(ctx.dest.url, src);
+        if (ctx.isIframe && ctx.dest.referer) {
+            var isCrossDomainOldLocation = !urlUtils.sameOriginCheck(ctx.dest.referer, ctx.dest.url);
```
Comment (miherlosev) [inline src/request-pipeline/header-transforms.js]: `isCrossDomainLocationBeforeRedirect`
`isCrossDomainLocationAfterRedirect`

File: test/server/proxy-test.js [hunk 2]
```diff
@@ -2147,5 +2152,58 @@ describe('Proxy', function () {
                 done();
             });
         });
+
+        it('Should correctly detect a cross-domain port of iframe after redirect (GH-1191)', function () {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `test('Should calculate a valid port for redirect in iframe(GH-...)')`

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by churkin** [comment]
We need to talk

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**